config/di.php

```diff
@@ -19,16 +19,22 @@
 use Yiisoft\Queue\Middleware\Push\MiddlewareFactoryPush;
 use Yiisoft\Queue\Middleware\Push\MiddlewareFactoryPushInterface;
 use Yiisoft\Queue\Middleware\Push\PushMiddlewareDispatcher;
+use Yiisoft\Queue\Provider\AdapterFactoryQueueProvider;
+use Yiisoft\Queue\Provider\QueueProviderInterface;
 use Yiisoft\Queue\Queue;
-use Yiisoft\Queue\QueueFactory;
-use Yiisoft\Queue\QueueFactoryInterface;
 use Yiisoft\Queue\QueueInterface;
 use Yiisoft\Queue\Worker\Worker as QueueWorker;
 use Yiisoft\Queue\Worker\WorkerInterface;
 
 /* @var array $params */
 
 return [
+    QueueProviderInterface::class => [
```

Comment:
Let's add definitions for all the providers, so if users want to use one of them, they can do it easily with just setting a definition for the provider interface.
```php
AdapterFactoryQueueProvider::class => [...],
QueueFactoryQueueProvider::class => [...],

// and set the default one
QueueProviderInterface::class => AdapterFactoryQueueProvider::class,
```

Reply:
Now the configuration is as it was before. I want to improve package configuration in one of the next PRs.

Reply:
There is no guarantee those future PRs will be merged into `master` soon. Those who already use this package should have a convenient configuration now. Let's do it before the PR is merged.

Reply:
Currently package in dev status. It's OK.

Reply:
For a not released package I'm for making a breaking change, but I'm against lack of configuration. I have two arguments for this:
- Some people already use this package. I'd prefer to make its usage as comfortable as we can. Despite there are breaking changes.
- If a change needs to have a default configuration, it should have a default configuration from the very beginning. This way we'll avoid decisions, which are good at architectural point, but are hard to be used in real-life projects. The goal is to provide some default configuration, document it and see if it's really comfortable to use in a project. If it's impossible to describe it in simple words - than the decision is likely to be reimagined and simplified for the end user.